okhttp-tests/src/test/java/okhttp3/internal/framed/Http2ConnectionTest.java

```diff
@@ -497,56 +483,1063 @@ private Buffer data(int byteCount) {
     FramedConnection connection = new FramedConnection.Builder(true)
         .socket(socket)
         .pushObserver(IGNORE)
-        .protocol(HTTP_2.getProtocol())
         .build();
     connection.start(false);
     socket.shutdownOutput();
     try {
-      connection.newStream(headerEntries("a", longString), false, true);
+      connection.newStream(headerEntries("a", longString), false);
       fail();
     } catch (IOException expected) {
     }
     try {
-      connection.newStream(headerEntries("b", longString), false, true);
+      connection.newStream(headerEntries("b", longString), false);
       fail();
     } catch (IOException expected) {
     }
   }
 
-  private FramedConnection sendHttp2SettingsAndCheckForAck(boolean client, Settings settings)
-      throws IOException, InterruptedException {
-    peer.setVariantAndClient(HTTP_2, client);
-    peer.sendFrame().settings(settings);
-    peer.acceptFrame(); // ACK
+  @Test public void clientCreatesStreamAndServerReplies() throws Exception {
```

Comment:
Probably the most technical part of this change was migrating tests from SPDY/3 to HTTP/2. We originally covered some codepaths only with SPDY because they’re so similar.